### ISSUE TITLE
[Feature]: Resource Manager

### ISSUE THREAD

**Issue by Ruan625Br** [AndroidIDEOfficial/AndroidIDE#1319]
### Feature description

**Resource Manager's Objective:**

The Resource Manager was added to simplify access to resources in the project. The initial approach included the following points:

1. Display a list of all resources from all project modules.
2. Allow the selection of multiple resources.
3. Show XML images from the "mipmap" and "drawable" folders.

However, after a review, the approach was adjusted as follows:

1. Display a list of resources only from the "app" module (for now).
2. Instead of directly displaying XML images from the "mipmap" and "drawable" folders, an alternative solution was implemented.

|  Structure | 
| ------------ | 
| ![Resource Manager Structure](https://github.com/AndroidIDEOfficial/AndroidIDE/assets/98613839/9e0215b0-1a46-453f-865b-cd3b8d1b3e70)

|  EditorActivityKt | ResourceListFragment - Drawable | ResourceListFragment - Color | ResourceListFragment - Layout|
| ------------ |  ------------ |  ------------ |  ------------ | 
|![img1](https://github.com/AndroidIDEOfficial/AndroidIDE/assets/98613839/e8deda73-bb71-4877-b80b-989d892d0456) | ![img2](https://github.com/AndroidIDEOfficial/AndroidIDE/assets/98613839/95efe27e-0f44-4d34-901e-036310cba5ab) | ![img3](https://github.com/AndroidIDEOfficial/AndroidIDE/assets/98613839/b58e59ef-e90e-4737-b8db-7a41032f3cf4)| ![img4](https://github.com/AndroidIDEOfficial/AndroidIDE/assets/98613839/fff16d9d-5866-4267-8eae-b792212da548)

Video on the official AndroidIDE channel on [Telegram](https://t.me/androidide_discussions/1/92326)

**Simple way to add new resource**
``` kotlin
  resourceInfoList.add(ResourceInfo(
      name = "Layout",
      file = File(layoutPath),
      resourceType = ResourceType.LAYOUT))
``` 

These changes aim to simplify the use of the Resource Manager and enhance the user experience when accessing and selecting resources in the project.

### What version of AndroidIDE you're using?

v2.5.2 (debug builds)

### Duplicate issues

- [X] This issue has not been reported yet.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by MrIkso** [comment]
I think add new activity not good idea. Try refactoring code , using a fragment

**Comment by Ruan625Br** [comment]
I wanted to create the ResourceManagerFragment for the SidebarAction, but I think there would be little space.

**Comment by Ruan625Br** [comment]
## Using fragment
<img src="https://github.com/AndroidIDEOfficial/AndroidIDE/assets/98613839/9d28803a-63c7-4efb-9cdb-3e2e8844617c" width="41%" height="41%">

**Comment by MrIkso** [comment]
Looking not bad

**Comment by Rafael2616** [comment]
Looks pretty cool..

**Comment by MatheusTGamerPro** [comment]
**Impressive.** In line with AndroidIDE's design principles, specifically adhering to Material Design guidelines, it's worth considering the addition of a *Floating Action Button (FAB)* within the parent layout of the `ResourceManagerFragment`, positioned independently of the individual resource fragments. This FAB could dynamically adapt its functionality based on the user's context. For instance, when the user navigates to the **Color** section, it could facilitate the creation of new color resources and seamlessly extend this capability to other relevant resource types.

Moreover, to address concerns related to screen real estate, an alternative approach could involve integrating a discrete action button, strategically positioned at the bottom right corner of the "Resource Manager" title within the view. This approach not only conserves screen space but also ensures accessibility and usability.

These recommendations are offered as creative solutions to enhance the user experience and align with modern design standards, aiming to elevate the overall functionality and aesthetics of the `ResourceManagerFragment`. *I hope this suggestion proves helpful.*

**Comment by Ruan625Br** [comment]
> **Impressionante.** De acordo com os princípios de design do AndroidIDE, aderindo especificamente às diretrizes de Design de Material, vale a pena considerar a adição de um _Botão de Ação Flutuante (FAB)_ dentro do layout pai do , posicionado independentemente dos fragmentos de recursos individuais. Essa FAB poderia adaptar dinamicamente sua funcionalidade com base no contexto do usuário. Por exemplo, quando o usuário navega para a seção **Cor**, isso pode facilitar a criação de novos recursos de cores e estender perfeitamente esse recurso a outros tipos de recursos relevantes.`ResourceManagerFragment`
> 
> Além disso, para resolver preocupações relacionadas a imóveis na tela, uma abordagem alternativa poderia envolver a integração de um botão de ação discreto, estrategicamente posicionado no canto inferior direito do título "Gerenciador de Recursos" dentro da exibição. Essa abordagem não apenas conserva o espaço da tela, mas também garante acessibilidade e usabilidade.
> 
> Essas recomendações são oferecidas como soluções criativas para melhorar a experiência do usuário e se alinhar aos padrões modernos de design, visando elevar a funcionalidade geral e a estética do . _Espero que esta sugestão seja útil._`ResourceManagerFragment`

I'll look into it, thanks for the opinion.

**Comment by itsaky** [comment]
@Ruan625Br Please make the layout more compact. We don't really need square previews of the colors. You could make the color previews circular or a rectangle with the height same as the two textviews (i.e. the bottom of the color preview should be aligned to the bottom of the second textview).

Also, instead of showing the color integers, it should show the HEX string for the colors.

**Comment by itsaky** [comment]
> I'll do that, I forgot to convert integers to hex.

Why convert integers to hex? The integers are defined as hex strings in `colors.xml` file, right?

**Comment by Ruan625Br** [comment]
Yes, but I don't read the **colors.xml** file directly, but rather all the **.xml** files inside the **values** folder

**Comment by Ruan625Br** [comment]
> Sim, mas eu não leio o arquivo .xml **cores** diretamente, mas sim todos os arquivos **.xml** dentro da pasta **de valores**

and after I refactor the code I will use this same way to read strings, colors... in an xml file with just one code

**Comment by Ruan625Br** [comment]
Interesting, this is the way I read colors:

 ``` kotlin
fun findAllColorsInXmlFiles(file: File): List<ColorInfo> {
     val results = mutableListOf<ColorInfo>()
     val log = ILogger.newInstance("ColorUtils")

     if (!file.isDirectory)
       return results

     val xmlFilePattern = Pattern.compile(".*\\.xml$")

     file.listFiles { _, name -> xmlFilePattern.matcher(name).matches() }
       ?.forEach { xmlFile ->
         try {
           val xmlContent = xmlFile.readText(Charset.defaultCharset())
           val pattern = "<color name=\"([^\"]+)\">#([A-Fa-f0-9]+)</color>".toPattern()
           val matcher = pattern.matcher(xmlContent)

           while (matcher.find()) {
             val colorName = matcher.group(1)
             val colorHex = matcher.group(2)
             val colorValue = Color.parseColor("#$colorHex")
             results.add(ColorInfo(xmlFile, colorName!!, colorValue))
           }
         } catch (e: Exception) {
           log.error("Failed to get colors", e)

         }
       }

     return results
   }
```
Initially I had in mind to create a data class to store the search result within the xml file, this way we could look for values not defined directly in the code, we would have reusable code.


---

After obtaining the colors, I convert the ColorInfo to ResourceItem:

``` kotlin
 if (type == ResourceType.COLOR) {
          val colorList = ColorUtils.findAllColorsInXmlFiles(path.toFile())
          colorList.map { resourceList.add(it.toResourceItem()) }
          Success(resourceList as List<ResourceItem>)

  }
```

Converter:

``` kotlin
fun ColorInfo.toResourceItem(): ResourceItem {
  return ResourceItem(
    name = colorName,
    info = colorValue.toString(),
    type = ResourceType.COLOR,
    file = file
  )
}
```


I intend to add path observer in the future, this way we would know when a file was changed, it will stay inside the ResourceListLiveData. This Observer is used by file managers (I used it myself in my file manager)

All of this occurs in the **ResourceListLiveData** and the result is sent to the **ResourceListMapLiveData** then we have access to the result through the **ResourceListViewModel**.
